pkg/controller/export/export_connection_controller.go

```diff
@@ -0,0 +1,320 @@
+
+	log.Info("Connection upsert successful", "name", request.Name, "driver", connObj.Spec.Driver)
+	return reconcile.Result{}, updateOrCreateConnectionPodStatus(ctx, r.reader, r.writer, r.scheme, connObj, []*statusv1alpha1.ConnectionError{}, nil, r.getPod)
+}
+
+func UpdateOrCreateConnectionPodStatus(
+	ctx context.Context,
+	reader client.Reader,
+	writer client.Writer,
+	scheme *runtime.Scheme,
+	connObjName string,
+	exportErrors []*statusv1alpha1.ConnectionError,
+	activeConnection *bool,
+	getPod func(context.Context) (*corev1.Pod, error)) error {
+
+	// Since the caller from Audit won't have an incoming request
+	// use the connection name from the audit connection flag as the predetermined connection name
+	request := types.NamespacedName{
+		Namespace: util.GetNamespace(),
+		Name:      connObjName,
+	}
+	connObj := &connectionv1alpha1.Connection{}
+	err := reader.Get(ctx, request, connObj)
+	if err != nil {
+		return err
+	}
+	return updateOrCreateConnectionPodStatus(ctx, reader, writer, scheme, connObj, exportErrors, activeConnection, getPod)
+}
+
+func updateOrCreateConnectionPodStatus(ctx context.Context,
+	reader client.Reader,
+	writer client.Writer,
+	scheme *runtime.Scheme,
+	connObj *connectionv1alpha1.Connection,
+	exportErrors []*statusv1alpha1.ConnectionError,
+	activeConnection *bool,
+	getPod func(context.Context) (*corev1.Pod, error)) error {
+
+	pod, err := getPod(ctx)
+	if err != nil {
+		return fmt.Errorf("getting reconciler pod: %w", err)
+	}
+
+	// Check if it exists already
+	statusNS := pod.Namespace
+	statusName, err := statusv1alpha1.KeyForConnection(pod.Name, connObj.GetNamespace(), connObj.GetName())
+	if err != nil {
+		return fmt.Errorf("getting key for connection: %w", err)
+	}
+	shouldCreate := true
+	connPodStatusObj := &statusv1alpha1.ConnectionPodStatus{}
+
+	err = reader.Get(ctx, types.NamespacedName{Namespace: statusNS, Name: statusName}, connPodStatusObj)
+
+	existingActiveConnection := false
+	switch {
+	case err == nil:
+		shouldCreate = false
+		// ConnectionPodStatus object exists so get the existing active state
+		existingActiveConnection = connPodStatusObj.Status.Active
+	case apierrors.IsNotFound(err):
+		if connPodStatusObj, err = newConnectionPodStatus(scheme, pod, connObj); err != nil {
+			return fmt.Errorf("creating new connection connPodStatusObj: %w", err)
+		}
+	default:
+		return fmt.Errorf("getting connection object status in name %s, namespace %s: %w", connObj.GetName(), connObj.GetNamespace(), err)
+	}
+
+	// nil indicates expected active Connection state is unknown by caller during Upsert
+	if activeConnection == nil && connPodStatusObj.Status.ObservedGeneration != connObj.GetGeneration() {
+		// Reset the active connection state when there are updates to the Connection object to ensure the active state is only true when the Publish succeeds for the current Connection
+		resetActiveConnection := false
+		activeConnection = &resetActiveConnection
+	} else if activeConnection == nil {
```

Comment:
when will `activeConnection == nil` be true? Just curious. 

In the code `activeConnection` is true if published messages are > 0 which is always either result in true or false, right?